### PR TITLE
Adding cursor-changed event to webContents

### PR DESCRIPTION
**Pull request by brenca**
This PR adds the aforementioned event to `webContents` by listening to the IPC messages. This could be useful because if the `cursor` css property is not explicitly set, you have no way to determine what the actual cursor type is, because chrome is the one that makes that decision. We use this with offscreen rendering to display the proper cursor over the rendered frames.

### CODE REVIEW

File: atom/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -1039,6 +1042,19 @@ void WebContents::EndFrameSubscription() {
     view->EndFrameSubscription();
 }
 
+void WebContents::OnCursorChange(const content::WebCursor& cursor) {
+  content::WebCursor::CursorInfo* info = new content::WebCursor::CursorInfo();
```
Comment (zcbenz) [inline atom/browser/api/atom_api_web_contents.cc]: The memory used by `info` is leaked, we can just allocate it on stack:

```cpp
content::WebCursor::CursorInfo info;
cursor.GetCursorInfo(&info);
```

File: docs/api/web-contents.md [hunk 2]
```diff
@@ -281,6 +281,28 @@ Emitted when a page's theme color changes. This is usually due to encountering a
 <meta name='theme-color' content='#ff0000'>
 ```
 
+### Event: 'cursor-changed'
+
+Returns:
+
+* `event` Event
+* `type` String
+* `image` NativeImage
```
Comment (zcbenz) [inline docs/api/web-contents.md]: `image` and `scale` should be marked as `(Optional)`.

File: atom/browser/api/atom_api_web_contents.cc (hunk 1 above)
Comment (brenca) [inline atom/browser/api/atom_api_web_contents.cc]: Whoops, yeah, sorry, gonna fix that in a moment.

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**